xarray/tests/test_backends.py

```diff
@@ -992,6 +999,46 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+class OpenMFDatasetTest(TestCase):
+    def validate_open_mfdataset_large_num_files(self, engine, nfiles=2000):
```

Comment:
Strange though, it is causing false positives on too many open file errors relative to using `nfiles=2000` from before- I'm not sure what is happening.  

Another downside here is needing to support linux/windows since `resource` is just *nix, not windows.  I think I'll revert to using fast / slow test cases for now because of these two issues.